garden_ai/entrypoints.py

```diff
@@ -342,7 +358,17 @@ def inner(*args, **kwargs):
             if os.environ.get("GARDEN_SKIP_TESTS") == str(True):
                 return None
             else:
-                return test_func(*args, **kwargs)
+                # call the test_func once
+                result = test_func(*args, **kwargs)
+                try:
+                    # Call the test_func again with the same args to enforce idempotency.
+                    if result != test_func(*args, **kwargs):
+                        raise EntrypointIdempotencyError(
+                            "Your entrypoint function is not indempotent."
+                        )
+                    return result
+                except Exception as e:
```

Comment:
I don't hate the pattern of raising an error from the try block just to except it immediately, but raising `EntrypointTestError() from e` when `e` could already be an `EntrypointIdempotencyError` is an unnecessary middleman imo

Reply:
I see what you are saying, and yeah, it doesn't make much sense to raise an error just to immediately catch it and throw another. My intention was to catch other errors that might occur when calling the entrypoint a second time, but I think it will be more informative to just let those propagate on their own.